components/tikv_util/src/sys/mod.rs

```diff
@@ -6,108 +6,87 @@ pub mod cpu_time;
-            }
-
-            if cpu_quota != 0. {
-                cpu_num = cpu_num.min(cpu_quota);
-            }
-
-            super::limit_cpu_cores_quota_by_env_var(cpu_num)
+        if cpu_quota != 0. {
+            cpu_num = cpu_num.min(cpu_quota);
         }
 
-        pub fn memory_limit_in_bytes(&self) -> u64 {
-            let total_mem = {
-                let mut system = SYS_INFO.lock().unwrap();
-                system.refresh_memory();
-                system.get_total_memory() * KIB
-            };
-            let cgroup_memory_limits = self.cgroup.memory_limit_in_bytes();
-            if cgroup_memory_limits <= 0 {
-                total_mem
-            } else {
-                std::cmp::min(total_mem, cgroup_memory_limits as u64)
-            }
-        }
-
-        pub fn log_quota(&self) {
-            info!(
-                "memory limit in bytes: {}, cpu cores quota: {}",
-                self.memory_limit_in_bytes(),
-                self.cpu_cores_quota()
-            );
-        }
+        limit_cpu_cores_quota_by_env_var(cpu_num)
     }
-}
-
-#[cfg(not(target_os = "linux"))]
-pub mod sys_quota {
-    use super::super::config::KIB;
-    use super::{SystemExt, SYS_INFO};
 
-    pub struct SysQuota {}
+    #[cfg(not(target_os = "linux"))]
+    pub fn cpu_cores_quota() -> f64 {
+        let cpu_num = num_cpus::get() as f64;
+        super::limit_cpu_cores_quota_by_env_var(cpu_num)
+    }
 
-    impl SysQuota {
-        pub fn new() -> Self {
-            Self {}
+    #[cfg(target_os = "linux")]
+    pub fn memory_limit_in_bytes() -> u64 {
+        let total_mem = Self::sysinfo_memory_limit_in_bytes();
+        let cgroup_memory_limit = SELF_CGROUP.memory_limit_in_bytes();
+        if cgroup_memory_limit <= 0 {
+            total_mem
+        } else {
+            std::cmp::min(total_mem, cgroup_memory_limit as u64)
         }
+    }
 
-        pub fn cpu_cores_quota(&self) -> f64 {
-            let cpu_num = num_cpus::get() as f64;
-            super::limit_cpu_cores_quota_by_env_var(cpu_num)
-        }
+    #[cfg(not(target_os = "linux"))]
+    pub fn memory_limit_in_bytes() -> u64 {
+        Self::sysinfo_memory_limit_in_bytes()
+    }
 
-        pub fn memory_limit_in_bytes(&self) -> u64 {
-            let mut system = SYS_INFO.lock().unwrap();
-            system.refresh_memory();
-            system.get_total_memory() * KIB
-        }
+    pub fn log_quota() {
+        info!(
+            "memory limit in bytes: {}, cpu cores quota: {}",
+            Self::memory_limit_in_bytes(),
+            Self::cpu_cores_quota()
+        );
+    }
 
-        pub fn log_quota(&self) {
-            info!(
-                "memory limit in bytes: {}, cpu cores quota: {}",
-                self.memory_limit_in_bytes(),
-                self.cpu_cores_quota()
-            );
-        }
+    fn sysinfo_memory_limit_in_bytes() -> u64 {
+        let system = sysinfo::System::new_with_specifics(RefreshKind::new().with_memory());
```

Comment:
should it reuse the global sysinfo?

Reply:
Refresh memory is very light and these functions are only called when validating `TiKvConfig` or environments. `SYS_INFO` is only for diagnoise now.
So both this and `record_global_memory_usage` doesn't reuse `SYS_INFO`.